erts/emulator/test/match_spec_SUITE.erl

```diff
@@ -364,6 +365,35 @@ test_6(Config) when is_list(Config) ->
 
     ok.
 
+stack_size_rec(0, RefSize) ->
+    {stack_size, RefSize} = erlang:process_info(self(), stack_size),
+    f5_test6();
+stack_size_rec(Num, RefSize) ->
+    stack_size_rec(Num - 1, RefSize),
+    ok.
+
+%% Test stack_size/0
+test_stack_size(Config) when is_list(Config) ->
+    % Smallint
+    Fun = fun() -> stack_size_rec(10, 16) end,
+    Pat = [{'_', [], [{message, {stack_size}}]}],
+    P = spawn(?MODULE, fixed_runner, [self(), Fun]),
+    erlang_trace(P, true, [call]),
+    erlang_trace_pattern({?MODULE, f2_test6, 1}, Pat, [local]),
+    erlang_trace_pattern({?MODULE, f1_test6, 0}, Pat, [local]),
+    collect(P, [{trace, P, call, {?MODULE, f2_test6, [f1]}, 16},
+                {trace, P, call, {?MODULE, f1_test6, []}, 17}]),
+
+    % Bigint
+    Fun2 = fun() -> stack_size_rec(10000000, 10000006) end,
```

Comment:
If my calculations are correct, this will not cover the bignum if-branch.

The largest positive small integer (on 32-bit) is `1 bsl (32-5)` ≈ 134M. And that would require a stack size of 0.5Gibyte.

I think we can skip the bignum test, it will just cause out of memory problems on our 32-bit machines trying grow such a large process stack.